pymc_marketing/mmm/base.py

```diff
@@ -30,37 +29,27 @@
 __all__ = ("BaseMMM", "MMM")
 
 
-class BaseMMM:
+class BaseMMM(ModelBuilder):
     model: pm.Model
 
     def __init__(
         self,
-        data: pd.DataFrame,
-        target_column: str,
         date_column: str,
         channel_columns: Union[List[str], Tuple[str]],
         channel_prior: Optional[TensorVariable] = None,
-        validate_data: bool = True,
+        model_config: Optional[Dict] = None,
+        sampler_config: Optional[Dict] = None,
```

Comment:
Over all the PR looks consistent except for this part. Why we should not specify the target name but we should specify the date column and channel columns?

Reply:
now target name needs to be specified on the class level. The main issue is that load() from model builder is a static function, so can't access the user-passed target_column. @property function output_var was created to address this issue, as with it I can split assign the data from fit_data (part of idata) into self.X and self.y

Reply:
also, correct me if I'm wrong but shouldn't y be a single column df, or a pd.Series? In that case it's not needed to specify which column is responsible for target, if there's only one

Reply:
In my opinion, because the parameters that MMMconfig would be holding are simply strings, it just adds an extra step for the user without real benefit in exchange. Parameter encapsulation might be a good idea, if gathering of those params would be a complex process with some preprocessing etc. spanning across multiple functions.

Reply:
and regarding target_column being removed from params: the fact that we require date_column and channel_columns is based on the real need, as otherwise we'd make very rigid class that hardcodes a lot of information. X has multiple columns and users should be allowed to call them inside their dataset as they wish. With y there's no such need, and including a parameter that has no real use inside the class would be just strange.

Reply:
Thanks for the clarification! The mmm config approach would be indeed useful if we do not simply use strings. 

In this case I'll suggest with the two open points (tests and mmm notebook) to see how it would look at the end. At the moment I do not see any blockers. Still I would like to hear from other devs with more experience with user-facing API design as my experience is rather limited.